test/modifiers/batch.test.js

```diff
@@ -0,0 +1,284 @@
+var expect = require('chai').expect;
+var _ = require('lodash');
+var through = require('through2');
+
+var getReadableStream = require('../_utilities/getReadableStream.js');
+var runBasicStreamTests = require('../_utilities/runBasicStreamTests.js');
+
+var batch = require('../../').batch;
+
+function buff(str) {
+  return new Buffer(str);
+}
+
+describe('[throttle]', function () {
+  var data = ['item1', new Buffer('item2'), 'item3', 'item4'];
+  var objData = [true, false, [1, 2, 3], 'string', 0, '11', 95.23, { obj: true }, _.noop];
+
+  function basicTests(batchOptions, expected, objExpected) {
+    runBasicStreamTests(data, objData, function (stream, objectMode, done) {
+      var actual = [];
+
+      stream
+        .pipe(batch(batchOptions))
+        .on('data', function (chunk) {
+          actual.push(chunk);
+        })
+        .on('error', done)
+        .on('end', function () {
+          if (objectMode) {
+            expect(actual).to.deep.equal(objExpected);
+          } else {
+            expect(actual).to.deep.equal(expected);
+          }
+
+          done();
+        });
+    });
+
+    it('writes all chunks as an array', function (done) {
+      var input = getReadableStream([1, 2, 3], {
+        objectMode: true
+      });
+
+      var chunksRead = false;
+
+      input
+        .pipe(batch(batchOptions))
+        .on('data', function (chunk) {
+          chunksRead = true;
+          expect(chunk).to.be.an('array');
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunksRead).to.equal(true);
+          done();
+        });
+    });
+
+    it('successfully ends if there is no data written', function (done) {
+      var input = getReadableStream([], {
+        objectMode: true
+      });
+
+      var chunksRead = false;
+
+      input
+        .pipe(batch(batchOptions))
+        .on('data', function (chunk) {
+          chunksRead = true;
+          expect(chunk).to.be.an('array');
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunksRead).to.equal(false);
+          done();
+        });
+    });
+
+  }
+
+  describe('when "time" is defined in options', function () {
+    var expected = [[buff('item1')], [buff('item2'), buff('item3'), buff('item4')]];
+    var objExpected = [
+      objData.slice(0, 1),
+      objData.slice(1)
+    ];
+
+    basicTests({ time: 5 }, expected, objExpected);
```

Comment:
Since this uses the time option, would these tests also be flaky? It seems like the best option here is to update all of these tests to use a stubbed clock so they're reliable.